doc/developer-guides/hld/sriov-hld.rst

```diff
@@ -0,0 +1,303 @@
+   State Array Offset register of the SR-IOV Capability block.
+
+For more details, please refer to *PCIe Base Specification r4.0 chapter
+9.3.3*
+
+SR-IOV Architecture In ACRN
+---------------------------
+
+.. figure:: images/sriov-image3.png
+   :align: center
+   :width: 900px
+   :name: SR-IOV-architecure-in-acrn
+
+   SR-IOV Architectural in ACRN
+
+1. Hypervisor detects SR-IOV capable PCIe device in physical PCI device
+   enumeration phase.
+
+2. Hypervisor intercepts the PF’s SR-IOV capability accessing to
+   enable/disable VF devices based on *VF\_ENABLE* state, and all
+   read/write requests for PF device are passthrough to PF physical
+   device.
+
+3. Hypervisor waits for 100ms after *VF\_ENABLE* is set and initialize
+   VF devices. The differences between normal passthrough device and
+   SR-IOV VF device are physical device detection, BARs and MSIx
+   initializations. Hypervisor uses *Subsystem Vendor ID* to detect
+   SR-IOV VF physical device instead *Vendor ID* since there is no valid
+   *Vendor ID* for SR-IOV VF physical device, the VF BARs are
+   initialized by its associated PF’s SR-IOV capabilities not PCI
+   standard BAR registers, MSIx mapping base address is also from PF’s
+   SR-IOV capabilities not PCI standard BAR registers.
+
+SR-IOV Passthrough VF Architecture In ACRN
+------------------------------------------
+
+.. figure:: images/sriov-image4.png
+   :align: center
+   :width: 900px
+   :name: SR-IOV-vf-passthrough
+
+   SR-IOV VF Passthrough Architecture In ACRN
+
+1. The SR-IOV VF device needs to bind PCI-stud driver instead
+   vendor-specific VF driver before passthrough the device.
+
+2. User configures acrn-dm boot parameter with passthrough SR-IOV VF
+   device. When UOS starts, acrn-dm invokes a hypercall to set the
+   *vdev-VF0* device in UOS.
+
+3. Hypervisor emulates *Device ID/Vendor ID* and *Memory Space Enable
+   (MSE*) in configuration space for an assigned SR-IOV VF device. The
+   assigned VF *Device ID* comes from its associated PF’s capability,
+   *Vendor ID* is same with PF’s *Vendor ID* and *MSE* is always set
+   when reading the SR-IOV VF device *CONTROL* register
+
+4. Vendor-specific VF driver in target VM probes the assigned SR-IOV VF
+   device.
+
+SR-IOV Initialization Flow
+--------------------------
+
+.. figure:: images/sriov-image5.png
+   :align: center
+   :width: 900px
+   :name: SR-IOV-init-flow
+
+   SR-IOV Initialization Flow
+
+When a SR-IOV capable device is initialized, all access to the
+configuration space will pass through to the physical device directly.
+SOS can identify all capabilities of the device from SR-IOV extended
+capability then create sysfs node for SR-IOV management.
+
+SR-IOV VF Enable Flow
+---------------------
+
+.. figure:: images/sriov-image6.png
+   :align: center
+   :width: 900px
+   :name: SR-IOV-enable-flow
+
+   SR-IOV VF Enable Flow
+
+Application enables n VF devices via a SR-IOV PF device *sysfs* node.
```

Comment:
What application are we talking about here? ``acrn-dm``, launch script, something else?